tests/BenchmarkDotNet.Tests/Running/BenchmarkConverterTests.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System.Linq;
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Running;
+using Xunit;
+
+namespace BenchmarkDotNet.Tests.Running
+{
+    public class BenchmarkConverterTests
+    {
+        /// <summary>
+        /// https://github.com/dotnet/BenchmarkDotNet/issues/495
+        /// </summary>
+        [Fact]
+        public void ReadsAttributesFromBaseClass()
+        {
+            Benchmark benchmark = BenchmarkConverter.TypeToBenchmarks(typeof(Derived)).Single();
+           
+            Assert.NotNull(benchmark);
+            Assert.NotNull(benchmark.Target);
+            Assert.NotNull(benchmark.Target.IterationSetupMethod);
+            Assert.NotNull(benchmark.Target.IterationCleanupMethod);
+            Assert.NotNull(benchmark.Target.GlobalCleanupMethod);
+            Assert.NotNull(benchmark.Target.GlobalSetupMethod);
+        }
```

Comment:
this benchmark should not only Assert that given method is not null, but also that given method points to derived one, not the base one